core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -137,18 +138,18 @@ public synchronized void shutdown() {
       return;
     }
     shutdownStatus = Status.UNAVAILABLE.withDescription("transport was requested to shut down");
-    notifyShutdown();
+    notifyShutdown(shutdownStatus);
```

Comment:
Yes, UNAVAILABLE is retryable, but it is "most likely a transient condition and may be corrected by retrying with a backoff." We want to distinguish between "shutdown and retry with backoff" from "shutdown and retry immediately."